### PR TITLE
[6211] Broken repository due to manually fixing or editing during conflict resolution

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6211

### CODE REVIEW

File: ui/app/src/components/ConfirmDialog/utils.ts [hunk 1]
```diff
@@ -37,6 +37,8 @@ export interface ConfirmDialogProps extends ConfirmDialogBaseProps, Omit<DialogP
   classes?: Partial<Record<ConfirmDialogStateClassKey, string>>;
   disableOkButton?: boolean;
   disableCancelButton?: boolean;
+  okButtonText?: string | ReactNode;
```
Comment (rart) [inline ui/app/src/components/ConfirmDialog/utils.ts]: Doesn't ReactNode include string?

File: ui/app/src/components/GitManagement/GitManagement.tsx [hunk 2]
```diff
@@ -168,6 +194,21 @@ export function GitManagement(props: GitManagementProps) {
...
+    };
+    if (repoStatusConflictDialog.open && repoStatus && !repoStatus.clean) {
+      window.onbeforeunload = handleBeforeUpload;
+    } else {
+      window.onbeforeunload = null;
+    }
+    return () => {
+      window.onbeforeunload = null;
+    };
+  }, [repoStatus, repoStatusConflictDialog.open]);
```
Comment (rart) [inline ui/app/src/components/GitManagement/GitManagement.tsx]: Wouldn't this show the dialog for everyone landing on Git? the idea was use the promise on pull success to show only for the puller that first received the conflicts

File: ui/app/src/components/GitManagement/GitManagement.tsx [hunk 3]
```diff
@@ -121,22 +123,46 @@ export function GitManagement(props: GitManagementProps) {
...
+            })
+          );
+          reject(response);
+        }
+      });
     });
   }, [dispatch, fetchRepoStatusReceiver, setRepoStatusState, siteId]);
 
+  const onPullSuccess = (result: PullResponse) => {
+    fetchRepoStatus();
```
Comment (rart) [inline ui/app/src/components/GitManagement/GitManagement.tsx:149]: Wasn't the idea to open the dialog here after the fetch first results in conflicts

File: ui/app/src/components/GitManagement/GitManagement.tsx [hunk 4]
```diff
@@ -121,22 +123,46 @@ export function GitManagement(props: GitManagementProps) {
...
+  };
+
+  const onPullError = (response) => {
```
Comment (rart) [inline ui/app/src/components/GitManagement/GitManagement.tsx:152]: When the pull results in conflicts, it throws? Why is the pull error opening the dialog?

File: ui/app/src/components/GitManagement/GitManagement.tsx [hunk 5]
```diff
@@ -168,6 +194,21 @@ export function GitManagement(props: GitManagementProps) {
...
+  useEffect(() => {
+    const handleBeforeUpload = (event) => {
+      event.preventDefault();
+      return (event.returnValue = '');
+    };
+    if (repoStatusConflictDialog.open && repoStatus && !repoStatus.clean) {
+      window.onbeforeunload = handleBeforeUpload;
+    } else {
+      window.onbeforeunload = null;
+    }
```
Comment (rart) [inline ui/app/src/components/GitManagement/GitManagement.tsx]: The before unload needs to be in the dialog logic

File: ui/app/src/components/GitManagement/RepoStatus/RepoStatusUI.tsx [hunk 6]
```diff
@@ -91,15 +93,31 @@ export function RepoStatusUI(props: RepoStatusUIProps) {
...
-              cancelText={formatMessage(messages.no)}
-              confirmText={formatMessage(messages.yes)}
-              confirmHelperText={formatMessage(messages.confirmHelper)}
-              onConfirm={onRevertPull}
-            />
-            <Button variant="outlined" className={classes.commitButton} onClick={onCommitClick}>
+            <DropDownMenu
+              onMenuItemClick={onBulkAction}
+              variant="outlined"
+              // TODO: i18n
```
Comment (rart) [inline ui/app/src/components/GitManagement/RepoStatus/RepoStatusUI.tsx]: Anything left TODO?

File: ui/app/src/components/GitManagement/RepoStatus/RepoStatusUI.tsx [hunk 7]
```diff
@@ -162,7 +180,7 @@ export function RepoStatusUI(props: RepoStatusUIProps) {
             </Grid>
           )}
           {status.uncommittedChanges.length > 0 && (
-            <Grid item md={12} xl={status.conflicting.length ? 6 : 12}>
+            <Grid item md={12} xl={status.conflicting.length ? 12 : 12}>
```
Comment (rart) [inline ui/app/src/components/GitManagement/RepoStatus/RepoStatusUI.tsx]: 🤨
```suggestion
            <Grid item md={12}>
```
Is the `md=12` even necessary?

File: ui/app/src/components/GitManagement/RepoStatusConflictDialog/RepoStatusConflictDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    onCommitSuccess: onCommitSuccessProp,
+    onConflictResolved,
+    onFailedPullCancelled,
+    ...dialogProps
+  } = props;
+  const isRepoClean = status?.clean ?? false;
+  const [openConfirmDialog, setOpenConfirmDialog] = useState(false);
+
+  const onCommitSuccess = (status) => {
+    props.onClose?.(null, null);
```
Comment (rart) [inline ui/app/src/components/GitManagement/RepoStatusConflictDialog/RepoStatusConflictDialog.tsx]: This is up to the consumer. Should do on commit success prop

File: ui/app/src/components/GitManagement/RepoStatusConflictDialog/RepoStatusConflictDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    onCommitSuccessProp?.(status);
+  };
+
+  const onRevertSuccess = () => {
+    setOpenConfirmDialog(false);
+    props.onClose?.(null, null);
```
Comment (rart) [inline ui/app/src/components/GitManagement/RepoStatusConflictDialog/RepoStatusConflictDialog.tsx]: Same

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
.

File: ui/app/src/components/GitManagement/GitManagement.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/GitManagement/GitManagement.tsx:149]: pullSuccess doesn't trigger on conflicts, pullError is the one that triggers

File: ui/app/src/components/GitManagement/GitManagement.tsx (hunk 4 above)
Comment (jvega190) [reply ui/app/src/components/GitManagement/GitManagement.tsx:152]: If there are conflicts, pullError triggers, so when there are conflicts, after doing the fetchStatus (that fetch was already there), **if** there are conflicts, then the dialog will open

File: ui/app/src/components/ConfirmDialog/utils.ts (hunk 1 above)
Comment (jvega190) [reply ui/app/src/components/ConfirmDialog/utils.ts]: it does, updating

**Review by rart — APPROVED** [review]

**Pull request merged**